### PR TITLE
added nil pointer check to buckets.go

### PR DESCRIPTION
**Pull request by jakepage91**
I added a couple of lines to check if the pointer (metricsBucketSizebytesOutput) is not nil, if the length of the slice is greater than zero and then check if there is an error.

### CODE REVIEW

File: providers/aws/s3/buckets.go [hunk 1]
```diff
@@ -62,8 +62,10 @@ func Buckets(ctx context.Context, client ProviderClient) ([]Resource, error) {
 			}
 
 			bucketSize := 0.0
-			if len(metricsBucketSizebytesOutput.Datapoints) > 0 {
-				bucketSize = *metricsBucketSizebytesOutput.Datapoints[0].Average
+			if metricsBucketSizebytesOutput != nil && len(metricsBucketSizebytesOutput.Datapoints) > 0 {
+				if err != nil {
```
Comment (mlabouardy) [inline providers/aws/s3/buckets.go:66]: this check isn't needed since we checked before

**Review by mlabouardy — CHANGES_REQUESTED** [review]

File: providers/aws/s3/buckets.go (hunk 1 above)
Comment (jakepage91) [reply providers/aws/s3/buckets.go:66]: Just line 66 isn't needed right?

File: providers/aws/s3/buckets.go [hunk 2]
```diff
@@ -62,8 +62,9 @@ func Buckets(ctx context.Context, client ProviderClient) ([]Resource, error) {
 			}
 
 			bucketSize := 0.0
-			if len(metricsBucketSizebytesOutput.Datapoints) > 0 {
-				bucketSize = *metricsBucketSizebytesOutput.Datapoints[0].Average
+			if metricsBucketSizebytesOutput != nil && len(metricsBucketSizebytesOutput.Datapoints) > 0 {
+					bucketSize = *metricsBucketSizebytesOutput.Datapoints[0].Average
+				}
```
Comment (mlabouardy) [inline providers/aws/s3/buckets.go:67]: you've an extra bracket here

**Review by mlabouardy — CHANGES_REQUESTED** [review]

File: providers/aws/s3/buckets.go (hunk 2 above)
Comment (jakepage91) [reply providers/aws/s3/buckets.go]: Removed!

**Review by mlabouardy — APPROVED** [review]

**Pull request merged**